src/use-swr.ts

```diff
@@ -192,6 +192,11 @@ function useSWR<Data = any, Error = any>(
     config
   )
 
+  const lastestConfig = useRef(config)
+  useIsomorphicLayoutEffect(() => {
```

Comment:
I'm not sure if we need to have this useEffect here, could it just be updated in each rendering directly?

```js
const configRef = useRef(config)
configRef.current = config
```

Reply:
@huozhi I feel that we still need to wrap it with `useEffect` (`useIsomorphicLayoutEffect` here). Updating refs directly in render can be dangerous in Concurrent Mode (not 100% about this specific usage here). 

I'll approve this for now, but I suggest adding a comment to clarify :+1:

Reply:
sure, good to know there's side effect on concurrent mode! 🙏 sorry for my unreasonable comment above

Reply:
@shuding thanks to point this out,do i need to change this back?

Reply:
yep, I think you can just simply revert the last commit. the rest looks good 👍